CorsixTH/Lua/world.lua

```diff
@@ -2880,3 +2880,14 @@ function World:setCampaignData(campaign_data)
   end
   self:getLocalPlayerHospital():setCampaignData(campaign_data.hospital)
 end
+
+--! Perform some checks on a map that a user is saving in the map editor
```

Comment:
It's a nice generic function that performs some checks and produce a warning if a problem is found.

Providing context outside this function in the Luadoc  (like your "a user is saving in the map editor") is tricky here, as in the future you or someone else may find another good use for these checks (eg a quick sanity check after loading a map, so the user gets a warning that it is not a good idea to play this map).  
If that happens, the Luadoc starts spreading false information (or at least partially incorrect information).

The safer option here is not to explain context of use for the function unless it's absolutely needed.
For example, if you have a number of functions that are designed to work together, and they only make sense in that way, (ie they are useless in any other situation). then it's fine to mention the usage context, because that's the one and only  way that the function(s) makes sense.

The latter  example seldomly happens, as one of the goals is to have many functions that are re-usable for various purposes..